### PR TITLE
Order dash dev

### PR DESCRIPTION
**Pull request by DanielmAaron**
FOODPRINT WHOLESALE - FOODPRINT-104
ORDER DASHBOARD

- [x] Pull weekly produce from sellerlogbooks
- [x] Display data in cards via produce rows
- [x] Pull weekly bids from buyerlogbooks
- [x] Display data with offers in produce categorised rows
- [x] Filter data (Produce Type, Province, Date)
- [x] Allow users to "order" by accepting offers or accepting bids

### CODE REVIEW

**Comment by DanielmAaron** [comment]
Screenshot 2022-06-02 at 11 08 13
Data binding to display data for available produce (being produce that match bid and offer produce)

File: routes/order.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+var sequelise = require('../config/db/db_sequelise');
+
+var models = initModels(sequelise);
+
+//GET order_dashboard
+router.get(
+  '/',
+  require('connect-ensure-login').ensureLoggedIn({ redirectTo: '/app/auth/login' }),
+  function (req, res, next) {
+    if (req.user.role === ROLES.Buyer || req.user.role === ROLES.Seller) {
```
Comment (jajukajulz) [inline routes/order.js:27]: also add admin or superuser here

File: routes/order.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+              'Parsley',
+              'Radish',
+              'Sorrel',
+              'Swiss Chard',
+              'Spinach',
+              'Turnips',
+            ];
+            const produceArray = [];
+            if (bid_rows.length || offer_rows.length) {
+              for (var i = 0; i < produce_rows.length; i++) {
```
Comment (jajukajulz) [inline routes/order.js:69]: lets make this more modular by creating a function that accepts the produce_rows, bid_rows and offer_rows and returns the finalProduceArray, then calling this function from inside your router.

i.e. something along the lines of - 
const finalProduceArray = []
`if (bid_rows.length || offer_rows.length) { 
finalProduceArray = returnRelevantProduce(produce_rows, bid_rows and offer_rows);
}

function returnRelevantProduce(produce_rows, bid_rows and offer_rows){
...
const relevantProduceArray = []
return relevantProduceArray = [...new Set(produceArray)];
}
`

**Comment by DanielmAaron** [comment]
### FINAL TESTING
As buyer or seller _add a bid or offer_
Then _view order dashboard_ (to see bid is viewable)
Login as **opposite role** to accept bid
See that bid is not viewable after _accepting_ order unless through **status filter**

### Other screenshots
Non-acceptable offer as viewing as a seller
Screenshot 2022-06-07 at 17 49 45

Settled Orders
Screenshot 2022-06-07 at 17 55 46

**Pull request merged**